Framework/Platform.cs

```diff
@@ -208,118 +201,117 @@ static unsafe Platform()
 	{
 		// This is done in this way so that the delegate is compatible with WASM/Emscripten,
 		// which do not accept normal delegates (only unmanaged function pointer ones like this) 
-		delegate* unmanaged[Cdecl]<IntPtr, int, void> fn = &HandleLog;
-		FosterSetLogCallback((IntPtr)fn, 0);
+		FosterSetLogCallback(&HandleLog, 0);
 	}
 
-	[DllImport(DLL)]
```

Comment:
What is the purpose of using `LibraryImport` instead of `DllImport`? `DllImport` has been working fine and not sure what the change is for, but I'm happy to use it if it is better somehow.

Reply:
It uses source generator and handles marshaling of out parameters and gives warnings if marshaling has some marshal issues